Mathlib/Data/Finset/Card.lean

```diff
@@ -734,6 +731,25 @@ theorem case_strong_induction_on [DecidableEq α] {p : Finset α → Prop} (s :
       (h₁ a s n) fun t ss => ih _ (lt_of_le_of_lt ss (ssubset_insert n) : t < _)
 #align finset.case_strong_induction_on Finset.case_strong_induction_on
 
+/-- Suppose that, given objects defined on all nonempty strict subsets of any nontrivial finset `s`,
+one knows how to define an object on `s`. Then one can inductively define an object on all finsets,
+starting from singletons and iterating. This can be used either to define data, or to prove
+properties.
+
+TODO: Actually let `p` be `Sort`-valued. -/
```

Comment:
What is blocking you from making this change now?

Reply:
I would need to replace `Finset.Nonempty.exists_eq_singleton_or_nontrivial` with computational content. I tried doing just that now, but I got stuck in Lean 4's `match` and induction intricacies.

Reply:
I suggest rewriting the docstring to record this.
```suggestion
starting from singletons and iterating.

TODO: Currently this can only be used to prove properties.
Replace `Finset.Nonempty.exists_eq_singleton_or_nontrivial` with computational content
in order to let `p` be `Sort`-valued. -/
```